src/Plugin/Bitbucket.php

```diff
@@ -0,0 +1,303 @@
+        $response = json_decode($response, true);
+
+        return (int)$response['id'];
+    }
+
+    /**
+     * @param int $commentId
+     * @param string $message
+     */
+    protected function createTaskForCommentInPullRequest($commentId, $message)
+    {
+        $this->apiRequest('/tasks', 'post', [
+            'anchor' => [
+                'id' => $commentId,
+                'type' => 'COMMENT',
+            ],
+            'text' => $message,
+        ]);
+    }
+
+
+    protected function updateBuild()
+    {
+        $endpoint = sprintf(
+            '/commits/%s',
+            $this->getBuild()->getCommitId()
+        );
+
+        switch ($this->getBuild()->getStatus()) {
+            case Build::STATUS_SUCCESS:
+                $state = 'SUCCESSFUL';
+                break;
+            case Build::STATUS_FAILED:
+                $state = 'FAILED';
+                break;
+            default:
+                $state = 'INPROGRESS';
+        }
+
+        $this->buildStatusRequest($endpoint, 'post', [
+            'state' => $state,
+            'key' => 'php-censor',
+            'name' => 'PHP Censor',
+            'url' => APP_URL . 'build/view/' . $this->getBuild()->getId(),
+            'description' => '',
+        ]);
+    }
+
+    /**
+     * @return Util\Plugin[]
+     */
+    protected function prepareResult()
+    {
+        $masterBuildStats = $this->findBuildErrorStats($this->findLatestBuild('master'));
+        $currentBuildStats = $this->findBuildErrorStats($this->findLatestBuild($this->build->getBranch()));
+
+        if (empty($masterBuildStats) && empty($currentBuildStats)) {
+            return [];
+        }
+
+        $plugins = array_unique(array_merge(array_keys($masterBuildStats), array_keys($currentBuildStats)));
+        sort($plugins);
+
+        $result = [];
+        foreach ($plugins as $plugin) {
+            $result[] = new Util\Plugin(
+                $plugin,
+                isset($masterBuildStats[$plugin]) ? $masterBuildStats[$plugin] : 0,
+                isset($currentBuildStats[$plugin]) ? $currentBuildStats[$plugin] : 0
+            );
+        }
+
+        return $result;
+    }
+
+    /**
+     * @param Util\Plugin[] $plugins
+     * @return array
+     */
+    protected function buildResultComparator(array $plugins)
+    {
+        $maxPluginNameLength = max(array_map('strlen', array_keys($plugins)));
+
+        $lines = [];
+        foreach ($plugins as $plugin) {
+            $lines[] = $plugin->generateFormatedOutput($maxPluginNameLength);
+        }
+
+        return $lines;
+    }
+
+    protected function reportGenerator(array $stats)
+    {
+        $message = str_replace(['%STATS%'], [implode(PHP_EOL, $stats)], $this->message);
+
+        return $this->builder->interpolate($message);
+    }
+
+    protected function findLatestBuild($branchName)
+    {
+        $query = 'SELECT max(id) FROM build WHERE project_id = :id AND branch = :branch_name';
```

Comment:
Could you use placeholders `{{name}}` instead of table/column names please (For correct supporting of PostreSQL and MySQL escaping)? See [example](https://github.com/php-censor/php-censor/blob/master/src/Store/BuildErrorStore.php#L285).